src/arch/InputHandler/InputHandler_SDL.cpp

```diff
@@ -0,0 +1,974 @@
+            break;
+        case JOY_BUTTON_17: btn = SDL_GAMEPAD_BUTTON_RIGHT_PADDLE2;
+            break;
+        case JOY_BUTTON_18: btn = SDL_GAMEPAD_BUTTON_LEFT_PADDLE2;
+            break;
+        case JOY_BUTTON_19: btn = SDL_GAMEPAD_BUTTON_MISC2;
+            break;
+        case JOY_BUTTON_20: btn = SDL_GAMEPAD_BUTTON_MISC3;
+            break;
+        case JOY_BUTTON_21: btn = SDL_GAMEPAD_BUTTON_MISC4;
+            break;
+        case JOY_BUTTON_22: btn = SDL_GAMEPAD_BUTTON_MISC5;
+            break;
+        case JOY_BUTTON_23: btn = SDL_GAMEPAD_BUTTON_MISC6;
+            break;
+        case JOY_BUTTON_24: btn = SDL_GAMEPAD_BUTTON_TOUCHPAD;
+            break;
+        case JOY_HAT_UP: btn = SDL_GAMEPAD_BUTTON_DPAD_UP;
+            break;
+        case JOY_HAT_DOWN: btn = SDL_GAMEPAD_BUTTON_DPAD_DOWN;
+            break;
+        case JOY_HAT_LEFT: btn = SDL_GAMEPAD_BUTTON_DPAD_LEFT;
+            break;
+        case JOY_HAT_RIGHT: btn = SDL_GAMEPAD_BUTTON_DPAD_RIGHT;
+            break;
+        case JOY_LEFT:
+            axisBtn = SDL_GAMEPAD_AXIS_LEFTX;
+            axisPositive = false;
+            break;
+        case JOY_RIGHT: axisBtn = SDL_GAMEPAD_AXIS_LEFTX;
+            break;
+        case JOY_UP:
+            axisBtn = SDL_GAMEPAD_AXIS_LEFTY;
+            axisPositive = false;
+            break;
+        case JOY_DOWN: axisBtn = SDL_GAMEPAD_AXIS_LEFTY;
+            break;
+        case JOY_LEFT_2:
+            axisBtn = SDL_GAMEPAD_AXIS_RIGHTX;
+            axisPositive = false;
+            break;
+        case JOY_RIGHT_2: axisBtn = SDL_GAMEPAD_AXIS_RIGHTX;
+            break;
+        case JOY_UP_2:
+            axisBtn = SDL_GAMEPAD_AXIS_RIGHTY;
+            axisPositive = false;
+            break;
+        case JOY_DOWN_2: axisBtn = SDL_GAMEPAD_AXIS_RIGHTY;
+            break;
+        default: break;
+        }
+
+        // Handle face buttons
+        switch (SDL_GetGamepadButtonLabel(pGamepad, btn))
+        {
+        case SDL_GAMEPAD_BUTTON_LABEL_A:
+            return ssprintf("A %s", deviceString);
+        case SDL_GAMEPAD_BUTTON_LABEL_B:
+            return ssprintf("B %s", deviceString);
+        case SDL_GAMEPAD_BUTTON_LABEL_X:
+            return ssprintf("X %s", deviceString);
+        case SDL_GAMEPAD_BUTTON_LABEL_Y:
+            return ssprintf("Y %s", deviceString);
+        case SDL_GAMEPAD_BUTTON_LABEL_CROSS:
+            return ssprintf("Cross %s", deviceString);
+        case SDL_GAMEPAD_BUTTON_LABEL_SQUARE:
+            return ssprintf("Square %s", deviceString);
+        case SDL_GAMEPAD_BUTTON_LABEL_CIRCLE:
+            return ssprintf("Circle %s", deviceString);
+        case SDL_GAMEPAD_BUTTON_LABEL_TRIANGLE:
+            return ssprintf("Triangle %s", deviceString);
+        default: break;
+        }
+
+        // Handle buttons (if any) - don't capture face buttons here!
+        switch (btn)
+        {
+        case SDL_GAMEPAD_BUTTON_START:
```

Comment:
For reference: https://github.com/libsdl-org/SDL/blob/main/include/SDL3/SDL_gamepad.h#L129